src/DataSet/AttributeDataSet.php

```diff
@@ -25,7 +25,9 @@ final class AttributeDataSet implements RulesProviderInterface, DataSetInterface
     public function __construct(object $baseAnnotatedObject, array $data = [])
     {
         $this->baseAnnotatedObject = $baseAnnotatedObject;
-        $this->data = $data;
+        $this->data = ($data === [] && is_subclass_of($baseAnnotatedObject, DataSetInterface::class))
```

Comment:
`instanceof` could be used.

Reply:
We support attributes in non-`DataSetInterface`.

Reply:
> instanceof could be used.

Done.

Reply:
> We support attributes in non-DataSetInterface.

This code doesn't affect attributes support. It's just a convenience:

```php
$result = $validator->validate(new AttributeDataSet($dataSet));
```
instead of
```php
$result = $validator->validate(new AttributeDataSet($dataSet, $data));
```